cardano-cli/test/cardano-cli-test/Test/Cli/Parser.hs

```diff
@@ -97,3 +108,31 @@ hprop_integral_pair_reader_negative = propertyOnce $ do
     case Parsec.runParser pairIntegralParsecParser () "" s of
       Left parsecError -> Left $ show parsecError
       Right x -> Right x
+
+-- | Execute me with:
+-- @cabal test cardano-cli-test --test-options '-p "/cddl tx envelope completeness/"'@
+hprop_test_cddl_tx_envelope_completeness :: Property
+hprop_test_cddl_tx_envelope_completeness = property $ do
+  AnyShelleyBasedEra era <- H.noteShowM . forAll $ Gen.element [minBound .. maxBound]
+  x <- forAll $ genTx era
+  shelleyBasedEraConstraints era $ do
+    let TextEnvelopeType d = textEnvelopeType (proxyToAsType (getProxy x))
+    H.note_ $ "Envelope type: " <> show d
+    assert $ elem (T.pack d) txTextEnvelopeTypes
+ where
+  getProxy :: forall a. a -> Proxy a
+  getProxy _ = Proxy
+
+-- | Execute me with:
+-- @cabal test cardano-cli-test --test-options '-p "/cddl tx witness envelope completeness/"'@
+hprop_test_cddl_tx_witness_envelope_completeness :: Property
+hprop_test_cddl_tx_witness_envelope_completeness = property $ do
+  AnyShelleyBasedEra era <- H.noteShowM . forAll $ Gen.element [minBound .. maxBound]
+  x <- forAll $ genCardanoKeyWitness era
+  shelleyBasedEraConstraints era $ do
```

Comment:
Same as in the test above.

Reply:
Removed them here: https://github.com/IntersectMBO/cardano-cli/pull/892/commits/85b74885cb02e907acec7ed36d91200d79121ca1